### ISSUE TITLE
Add a bind method for GRPC

### ISSUE THREAD

**Issue by adriendomoison** [gin-gonic/gin#2031]
Hi,

GRPC is becoming a new standard for communication in between microservices and I was wondering if it would be possible to add to `c.BindProtobuf()` method to accommodate this new use case?

Here is an architecture example:
```
[Web] <-- REST API --> [Gateway] <-- GRPC --> [Microservices]
```

The problem is that the struct generated by GRPC does not have a `form` tag to map data. This makes `c.Bind()` not usable in this case.

The only solution at this point is to create a new struct, use it to bind the incoming data, and copy the content of the struct to the GRPC struct.

Adding a `bindProtobuf` method would allow mapping directly the incoming data to the structures generated by GRPC -> `c.BindProtobuf(&protopb.User{})`

This could be done by using the `protobuf.name`'s value generated by GRPC instead of the `form`'s value used in most `c.Bind()` methods:
```
type User struct {
	Username string `protobuf:"varint,1,opt,name=username,proto3" json:"username,omitempty"`
}
```

Would that be possible or is it out of scope?

**Comment by sinhashubham95** [comment]
I don't think this functionality is required. What you ultimately want to do is bind the request to a struct which is possible using c.BindJSON(&protopb.User{}).

**Comment by adriendomoison** [comment]
@sinhashubham95 I don't believe c.BindJSON(&protopb.User{}) is able to bind query parameters? How would you bind username with a request like `GET http://example.com/user?username=abc` ?
